### ISSUE TITLE
`magit-merge-into` gives "Wrong type argument: stringp, nil"

### ISSUE THREAD

**Issue by grdryn** [magit/magit#3550]
This used to work fine for me, but recently it has started failing (I think I've upgraded magit a couple of times since it started failing for me). All I'm doing is trying `m i` in my magit buffer as before (or `M-x magit-merge-into`). It fails with (from Messages buffer):

> cond: Wrong type argument: stringp, nil

My current version is:
`Magit 20180808.118, Git 2.17.1, Emacs 26.1, gnu/linux`

**Comment by grdryn** [comment]
New information! :)

I've just tried `m i` on the master branch of a repo, and it didn't give the condition (It asks what branch it should merge into...I don't want to merge master into anything though, so I cancel it). If I change back to the branch that I created with `magit-branch-spinoff`, I get the condition again.

My spinoff branch didn't have an upstream branch set. If I set that to master (that's what it suggests, saying it's the default), then I don't get the condition (I instead get a prompt for the branch that I want to merge into (default master)).

I believe this used to previously give me the prompt, even when I didn't have an upstream branch set (and that sounds less broken than the current behaviour).
